sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -81,16 +82,35 @@ export class ApplicationAESTController extends BaseController {
         `Application id ${applicationId} was not found.`,
       );
     }
-
+    let currentReadOnlyData: ApplicationFormData;
+    let previousReadOnlyData: ApplicationFormData;
     if (loadDynamicData) {
-      application.data =
-        await this.applicationControllerService.generateApplicationFormData(
+      const currentReadOnlyDataPromise =
+        this.applicationControllerService.generateApplicationFormData(
           application.data,
         );
+      // Check if a previous application exists.
+      const [previousApplicationVersion] =
+        await this.applicationService.getPreviousApplicationVersions(
+          applicationId,
+          { loadDynamicData: true, limit: 1 },
+        );
```

Comment:
Minor. Can this part which is blocking can be executed first, so that then next 2 request which are parallel can happen after this. ?

Reply:
I am not sure if I am following, to be clear, `generateApplicationFormData` happens first and does not block the `getPreviousApplicationVersions` to happen, hence both are already concurrent. 
`Promise.all` will then raise one more query and ensure both promises are resolved. What would be the benefit of the proposed change?

Reply:
It may be my feeling but i felt little more convenient to see both parallel requests one after the other instead of one awaiting request in the middle that's all. As I mentioned earlier it is not a blocker as there is no difference overall from execution perspective.

Reply:
I will make the change believing that it will be easier to understand in the future.

Reply:
Code is updated.